var/spack/repos/builtin/packages/charm/package.py

```diff
@@ -115,16 +164,19 @@ def install(self, spec, prefix):
         # We assume that Spack's compiler wrappers make this work. If
         # not, then we need to query the compiler vendor from Spack
         # here.
-        compiler = os.path.basename(self.compiler.cc)
-
-        options = [compiler]
-        if compiler == 'icc':
-            options.append('ifort')
-
-        options.extend([
-            "--with-production",   # Note: turn this into a variant
-            "-j%d" % make_jobs,
-            "--destination=%s" % prefix])
+        options = [
+            os.path.basename(self.compiler.cc),
+            os.path.basename(self.compiler.fc)
```

Comment:
Any reason not to use the full path to Spack's compiler wrappers? You can access this via `spack_cc` and `spack_fc`. Or does it use the compiler name to determine the compiler collection?

Reply:
Yes, after checking again, it seems like the compiler name is needed, not the full path. The build script uses that to determine which compiler to use.